### ISSUE TITLE
Strava activities w. Emojis and '@' fail to Dropbox, Runkeeper

### ISSUE THREAD

**Issue by bernardverhoeven** [cpfair/tapiriik#453]
Activities with emojis and @-charactars seem to fail when synced to Dropbox and Runkeeper (and maybe also to other services that don't get these characters/ don't allow them in or at the start of names).

Examples of names not syncing;
![image](https://user-images.githubusercontent.com/18517269/43584607-2245ee3e-9663-11e8-82a1-84222cc5f191.png)

I also have some sync errors for runs wih only '@' and also not only at the start.

For dropbox i found: 
https://www.dropbox.com/help/syncing-uploads/files-not-syncing
which refers to: 
"When naming your files, avoid characters incompatible with the Windows file system": https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file#naming-conventions

In short; maybe just removing everything non-standard should solve it?

Hope someone has time to figure this out as my knowledge of Python isn't sufficient of fixing the " def _format_file_name(self, format, activity):" in the dropbox.py source file.

Thanks!

**Comment by MattMedina** [comment]
Hi bernard,

I just found a quick fix that’s acceptable to me, I don’t know if it would be ok for you because it involves losing the name of the activity.

I changed the output filename structure in the Dropbox “setup” section of the tool. Just deleted the <NAME> property and syncing went fine.

I don’t use runkeeper but I guess you could then just resync from Dropbox to runkeeper.

**Comment by bernardverhoeven** [comment]
Hi Matt,

Thanks for reaching out!

Loosing the title is not ideal, but syncing is better than not-syncing, so
this would be great.

Is this effective immediately on the online version?

Thank you so much for working on this.

Regards, Bernard

--
Bernard Verhoeven
Fransen van de Puttelaan 32
3707 EH Zeist

✆ +31 6 8333 00 94

**Comment by MattMedina** [comment]
Hi Bernard,

I'm not working on the tool, I'm just a guy like you who happened to have the same issue with DB ;-)
While looking for a solution online I found your topic, and when I managed to work out a turnaround I came back to share with you.

The solution I suggested (removing the "name" variable frome the output filename in the dropbox setup window) did work for me the day that I posted.

Here are some screenshots of the process.

Click on "reconfigure" on the Dropbox box.
Then click on "setup"
Edit the text inside "Upload new activities as:" to remove the _NAME variable.
Hit save and sync again.


![2018-08-16 16_52_10-dashboard tapiriik](https://user-images.githubusercontent.com/42386253/44216361-04e28600-a175-11e8-809b-64099c0948f1.png)
![2018-08-16 16_52_40-dashboard tapiriik](https://user-images.githubusercontent.com/42386253/44216362-04e28600-a175-11e8-9a8f-0254dd239bb8.png)
![2018-08-16 16_56_45-dashboard tapiriik](https://user-images.githubusercontent.com/42386253/44216514-64409600-a175-11e8-85d9-f8c492b5ca9f.png)

**Comment by cpfair** [comment]
I wasn't able to reproduce the issue with Runkeeper (I was able to upload similarly emoji-laden activities without issue) - so that may be another problem.
